Buildings/Fluid/Dehumidifiers/Desiccant/BaseClasses/Performance.mo

```diff
@@ -0,0 +1,301 @@
+  Buildings.Fluid.Dehumidifiers.Desiccant.Data.Generic</a>,
+  <code>TProEnt</code> and <code>X_w_ProEnt</code> are the temperature and the humidity ratio of the process air entering the dehumidifier,
+  respectively.
+  <ul>
+     <li>
+     If <code>VReg_flow>VReg_flow_nominal</code>, the calculation is terminated and an error is generated.
+     </li>
+  </ul>
+  </li>
+  <li>
+  The temperature of the process air leaving the dehumidifier, <code>TProLea</code> is calculated by
+  <p align=\"center\" style=\"font-style:italic;\">
+    TProLea = TProEnt + (f(TProEnt,X_w_ProEnt,vPro,coeTProLea) + 273.15 - TProEnt)*etaSen,
+  </p>
+  where <code>coeTProLea</code> are coefficients.
+  <br> 
+  The humidity ratio of the process air leaving the dehumidifier, <code>X_w_ProLea</code>, is calculated by
+  <p align=\"center\" style=\"font-style:italic;\">
+    X_w_ProLea = X_w_ProEnt - (X_w_ProEnt - f(TProEnt,X_w_ProEnt,vPro,coeX_w_ProLea))*etaLat,
+  </p>
+  where <code>coeX_w_ProLea</code> are coefficients.
+  <ul>
+     <li>
+     If <code>X_w_ProLea</code> is less than 0,
+     the calculation is terminated and an error is generated.
+     </li>
+  </ul>
+  </li>
+  <li>
+  The specific heat of the regeneration, <code>CpReg</code> is calculated by
+  <p align=\"center\" style=\"font-style:italic;\">
+    CpReg =max(0,f(TProEnt,X_w_ProEnt,vPro,coeCpReg)*uSpe),
+  </p>
+  where <code>coeCpReg</code> are coefficients.
+  <br>
+  After that, the regeneration heating output ratio, <code>yQReg</code>, is calculated by
+  <p align=\"center\" style=\"font-style:italic;\">
+   yQReg = CpReg*(X_w_ProEnt-X_w_ProLea)*mPro_flow*(TRegEnt_nominal - TRegEnt) / (TRegEnt_nominal - TProEnt)/
+   QReg_flow_nominal,
+  </p> 
+  where <code>TRegEnt_nominal</code> and <code>TRegEnt</code> are the nominal temperature and the actual temperature
+  of the regeneration air entering the dehumidifier.<code>QReg_flow_nominal</code> is the nominal regeneration 
+  heating capacity.
+  <ul>
+     <li>
+     If <code>yQReg</code> is larger than 1,
+     the calculation is terminated and an error is generated.
+     </li>
+  </ul>
+  </li>
+ </ul>
+ </li>
+
+ <li>
+ Otherwise, 
+ <ul>
+  <li>
+  The outlet condition of the process air is the same as the inlet condition of the process air.
+  </li>
+  <li>
+  <code>VReg_flow</code> and <code>yQReg</code> are set to be 0.
+  </li>
+ </ul>
+ </li>
+
+</ul>
+<h4>References</h4>
+<ul>
+<li>
+<a href=\"https://energyplus.net/assets/nrel_custom/pdfs/pdfs_v22.1.0/EngineeringReference.pdf\">
+EnergyPlus v22.1.0 Engineering Reference</a>
+</li>
+<li>
+<a href=\"https://www.ashrae.org/file%20library/technical%20resources/covid-19/i-p_s20_ch26.pdf\">
+ASHRAE Handbook—HVAC Systems &amp; Equipment Chapter 26</a>
+</li>
+</ul>
+ 
+</html>", revisions="<html>
+<ul>
+<li>April 10, 2024, by Sen Huang:<br/>Added wheel speed ratio as an input. </li>
```

Comment:
do not need to add the second revision record, as it is still the first implementation.

Reply:
Removed the revision record.